File: DotNetGotchas/CSharp/RTTI/Modified3/RTTIExample/Dog.cs
//Dog.cs
using System;

namespace RTTIExample
{
	public class Dog : Animal
	{
		public void Bark()
		{
			Console.WriteLine("Dog barking");
		}

		public override void MakeNoise()
		{
			Bark();
		}

		public override bool Play(Animal other)
		{
			if (other is Dog) 
				return true;
			else
                return false;
		}

	}
}

File: DotNetGotchas/VBDN/RTTI/Modified3/RTTIExample/Dog.vb
'Dog.vb

Public Class Dog
    Inherits Animal

    Public Sub Bark()
        Console.WriteLine("Dog barking")
    End Sub


    Public Overrides Sub MakeNoise()
        Bark()
    End Sub

    Public Overrides Function Play(ByVal other As Animal) As Boolean
        If TypeOf other Is Dog Then
            Return True
        Else
            Return False
        End If
    End Function
End Class
